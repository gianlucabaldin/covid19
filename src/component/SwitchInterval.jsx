import React from 'react';
import { withStyles, Switch, Typography, Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const AntSwitch = withStyles((theme) => ({
  root: {
    width: 28,
    height: 16,
    padding: 0,
    display: 'flex',
  },
  switchBase: {
    padding: 2,
    color: theme.palette.grey[500],
    '&$checked': {
      transform: 'translateX(12px)',
      color: theme.palette.common.white,
      '& + $track': {
        opacity: 1,
        backgroundColor: theme.palette.primary.main,
        borderColor: theme.palette.primary.main,
      },
    },
  },
  thumb: {
    width: 12,
    height: 12,
    boxShadow: 'none',
  },
  track: {
    border: `1px solid ${theme.palette.grey[500]}`,
    borderRadius: 16 / 2,
    opacity: 1,
    backgroundColor: theme.palette.common.white,
  },
  checked: {},
}))(Switch);

const SwitchInterval = ({ onToggleSwitch, checked }) => {
  const { t } = useTranslation();
  const onToggle = (e) => {
    const newState = e.target.checked;
    onToggleSwitch(newState);
  };

  return (
    <Typography component="div">
      <Grid component="label" container alignItems="center" spacing={1}>
        <Grid item>{t('switch.historical')}</Grid>
        <Grid item>
          <AntSwitch checked={checked} onChange={onToggle} name="checkedC" />
        </Grid>
        <Grid item>{t('switch.last-10-days')}</Grid>
      </Grid>
    </Typography>
  );
};

export default SwitchInterval;
