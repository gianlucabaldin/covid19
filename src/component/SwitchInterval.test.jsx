import React from 'react';
import { shallow } from 'enzyme';
import SwitchInterval from './SwitchInterval';

describe('SwitchInterval', () => {
  const wrapper = shallow(<SwitchInterval />);
  it('should renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
