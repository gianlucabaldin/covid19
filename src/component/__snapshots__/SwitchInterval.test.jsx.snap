// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInterval should renders properly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="div"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    component="label"
    container={true}
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      Historical
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(WithStyles(ForwardRef(Switch)))
        name="checkedC"
        onChange={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      Last 10 days
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Typography))>
`;
